source/drivers/MicroBitFile.cpp

```diff
@@ -0,0 +1,280 @@
+
+    buff[size] = 0;
+
+    int ret = read(buff, size);
+
+    if(ret < 0)
+        return ManagedString();
+
+    return ManagedString(buff,ret);
+}
+
+/**
+  * Removes this MicroBitFile from the MicroBitFileSystem.
+  *
+  * @return MICROBIT_OK on success, MICROBIT_INVALID_PARAMETER if the given filename
+  *         does not exist, MICROBIT_CANCELLED if something went wrong.
+  */
+int MicroBitFile::remove()
+{
+    if(fileHandle < 0)
+        return MICROBIT_NOT_SUPPORTED;
+
+    int ret = MicroBitFileSystem::defaultFileSystem->remove(fileName.toCharArray());
+
+    if(ret < 0)
+        return ret;
+
+    fileHandle = MICROBIT_NOT_SUPPORTED;
+
+    return ret;
+}
+
+/**
+  * Seeks to the end of the file, and appends the given ManagedString to this MicroBitFile instance.
+  *
+  * @param bytes a pointer to the bytes to write to this file.
+  *
+  * @param len the number of bytes to write.
+  *
+  * @return The number of bytes written, MICROBIT_NOT_SUPPORTED if the current file
+  *         handle is invalid or this file was not opened in WRITE mode.
+  */
+int MicroBitFile::append(const char *bytes, int len)
+{
+    if(fileHandle < 0)
+        return MICROBIT_NOT_SUPPORTED;
+
+    int ret =  MicroBitFileSystem::defaultFileSystem->seek(fileHandle, 0, MB_SEEK_END);
+
+    if(ret < 0)
+        return ret;
+
+    return write(bytes,len);
+}
+
+/**
+  * Seeks to the end of the file, and appends the given ManagedString to this MicroBitFile instance.
+  *
+  * @param s The ManagedString to write to this file.
+  *
+  * @return The number of bytes written, MICROBIT_NOT_SUPPORTED if the current file
+  *         handle is invalid or this file was not opened in WRITE mode.
+  */
+int MicroBitFile::append(ManagedString s)
+{
+    return append(s.toCharArray(), s.length());
+}
+
+/**
+  * Returns the handle used by this MicroBitFile instance.
+  *
+  * @note This member function will also inform the user of any errors encountered
+  *       during the opening of this MicroBitFile. At open, the handle is set
+  *       to the return value from MicroBitFileSystem.open().
+  */
+int MicroBitFile::getHandle()
+{
+    return fileHandle;
+}
+
+/**
+  * Opens this MicroBitFile instance if the file has previously been closed.
+  *
+  * @param mode One of: READ, WRITE, READ_AND_WRITE. Defaults to READ_AND_WRITE.
+  *
+  * @return MICROBIT_OK on success, MICROBIT_NOT_SUPPORTED if the file is already open,
+  *         MICROBIT_INVALID_PARAMETER if the filename is too large, MICROBIT_NO_RESOURCES
+  *         if the file system is full, or memory could not be allocated..
+  *
+  * @note MicroBitFiles are opened at construction and are implicitly closed at
+  *       destruction. They can be closed explicitly using the close() member function.
+  */
+int MicroBitFile::open(int mode)
+{
+    if(fileHandle >= 0)
+        return MICROBIT_NOT_SUPPORTED;
+
+    int ret = MicroBitFileSystem::defaultFileSystem->open(fileName.toCharArray(), mode | CREATE);
```

Comment:
@jamesadevine But .. .open() doesn't take a filename, just a mode.  So you can't do that.  It only allows you to re-open a file that was named in the constructor.